engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -1716,6 +1720,22 @@ struct dark_thought_t final : public priest_buff_t<buff_t>
       }
     }
 
+    // TODO: check if you can have multiple out at once
+    if ( priest().sets->has_set_bonus( PRIEST_SHADOW, T28, B4 ) )
+    {
+      priest().procs.living_shadow->occur();
+
+      if ( priest().pets.your_shadow.n_active_pets() < 1 )
```

Comment:
You could simplify this check by just calling `auto pet = priest().pets.your_shadow.active_pet();` and then checking if pet is null or not.